coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/parsetree/ParseTreeBuilder.java

```diff
@@ -222,38 +222,55 @@ private SqlNode visitLateralViewInternal(ASTNode node, ParseContext ctx, boolean
-      // smart and for unnest(array[null]) determines return type to be null
-      SqlNode arrOfNull = SqlStdOperatorTable.ARRAY_VALUE_CONSTRUCTOR.createCall(ZERO, SqlLiteral.createNull(ZERO));
+      // array of [null] or map of (null, null) should be 3rd param to if function. With our type inference, calcite acts
+      // smart and for unnest(array[null]) or unnest(map(null, null)) determines return type to be null
+      SqlNode arrayOrMapOfNull;
+      if (operandCount == 3) {
+        arrayOrMapOfNull = SqlStdOperatorTable.ARRAY_VALUE_CONSTRUCTOR.createCall(ZERO, SqlLiteral.createNull(ZERO));
+      } else {
+        arrayOrMapOfNull = SqlStdOperatorTable.MAP_VALUE_CONSTRUCTOR.createCall(ZERO, SqlLiteral.createNull(ZERO),
+            SqlLiteral.createNull(ZERO));
+      }
       HiveFunction hiveIfFunction = functionResolver.tryResolve("if", null, 1);
-      SqlCall ifFunctionCall = hiveIfFunction.createCall(SqlLiteral.createCharString("if", ZERO),
-          ImmutableList.of(ifCondition, unnestOperand, arrOfNull), null);
-      unnestOperand = ifFunctionCall;
+      unnestOperand = hiveIfFunction.createCall(SqlLiteral.createCharString("if", ZERO),
+          ImmutableList.of(ifCondition, unnestOperand, arrayOrMapOfNull), null);
+    }
+    if (operandCount == 3) { // unnest explode array
+      unnestCall = HiveExplodeOperator.EXPLODE.createCall(ZERO,
+          SqlStdOperatorTable.AS.createCall(ZERO, unnestOperand, aliasOperands.get(2)));
+      SqlNode unnestAlias =
+          SqlStdOperatorTable.AS.createCall(ZERO, unnestCall, aliasOperands.get(1), aliasOperands.get(2));
+      SqlNode rightSelect = new SqlSelect(ZERO, null, new SqlNodeList(ImmutableList.of(SqlIdentifier.star(ZERO)), ZERO),
+          unnestAlias, null, null, null, null, null, null, null);
+      SqlNode lateralCall = SqlStdOperatorTable.LATERAL.createCall(ZERO, rightSelect);
+      SqlCall aliasCall =
+          SqlStdOperatorTable.AS.createCall(ZERO, lateralCall, aliasOperands.get(1), aliasOperands.get(2));
+      return new SqlJoin(ZERO, sqlNodes.get(1), SqlLiteral.createBoolean(false, ZERO), JoinType.COMMA.symbol(ZERO),
+          aliasCall/*lateralCall*/, JoinConditionType.NONE.symbol(ZERO), null);
+    } else { // unnest explode map
+      unnestCall = HiveExplodeOperator.EXPLODE.createCall(ZERO, unnestOperand);
+      SqlNode rightSelect = new SqlSelect(ZERO, null, new SqlNodeList(ImmutableList.of(SqlIdentifier.star(ZERO)), ZERO),
+          unnestCall, null, null, null, null, null, null, null);
+      SqlNode lateralCall = SqlStdOperatorTable.LATERAL.createCall(ZERO, rightSelect);
+      SqlCall aliasCall = SqlStdOperatorTable.AS.createCall(ZERO, lateralCall, aliasOperands.get(1),
+          aliasOperands.get(2), aliasOperands.get(3));
+      return new SqlJoin(ZERO, sqlNodes.get(1), SqlLiteral.createBoolean(false, ZERO), JoinType.COMMA.symbol(ZERO),
+          aliasCall/*lateralCall*/, JoinConditionType.NONE.symbol(ZERO), null);
```

Comment:
this block is identical to block lines 257-263, can we combine them after the if statement?